folding/src/checker.rs

```diff
@@ -0,0 +1,342 @@
+//! A kind of pseudo-prover, will compute the expressions over the witness a check row by row
```

Comment:
Is it expected in this PR?

Reply:
Oh shoot, why is this commit in this PR... No, this is meant to be in https://github.com/o1-labs/proof-systems/pull/2181